File: defaultcode/cpp/observer/observer.cpp
#include <iostream>
#include <vector>
using namespace std;

//subject class
class $classname {
    vector<Observer*> observers; 

  public:

    void attach(Observer *obs) {
        observers.push_back(obs);
    }

    void notify(){
        for (int i = 0; i < observers.size(); i++)
            observers[i]->update();
    }
};

//observer class
class Observer {

    $classname *model;
    
  public:
    Observer($classname *mod) {
        model = mod;
        model->attach(this);
    }

    void update(){

    }
};

File: defaultcode/Java/observer/observer.java
//subject class
public class $classname {
    List<Observer> observers; 

    void attach(Observer obs) {
        observers.add(obs);
    }

    void notify(){
        for (int i=0; i<observers.size(); i++){
            observers[i].update();
        }
            
    }
}

//observer class
class Observer {

    public $classname model;
    

    public Observer($classname mod) {
        model = mod;
        model.attach(this);
    }

    public void update(){

    }
}